### PR TITLE
ILABS-142 Added all new taxonomies in basic form (some need additiona…

### PR DESCRIPTION
**Pull request by sam-ludlow**
…l properties)
The git diff here looks a total mess, if you manually diff:
https://raw.githubusercontent.com/invotra/api/SWAGGER-4.36/api.swagger.yaml
with
https://raw.githubusercontent.com/invotra/api/ILABS-142.SWAGGER-4.36/api.swagger.yaml
it looks fine.

### CODE REVIEW

**Comment by ParryLuke** [comment]
We should also look to add any properties (has_children, weight, maybe others) that are required to build the new taxonomy screens (I'm not actually sure which ones exactly are needed) to the existing taxonomies: teams, locations, job roles. We can only add properties to these existing endpoints, don't change/remove any existing properties or parameters as that would be a breaking change.

Don't add has_children for job roles even if it is required for teams/locations. As they shouldn't have a hierarchy, I'll raise a task to remove parent_uuid from job roles.

File: api.swagger.yaml [hunk 1]
```diff
@@ -2392,172 +2392,3580 @@ paths:
...
+          description: 'The badge was successfully deleted.'
+        '400':
+          description: 'The supplied UUID was malformed.'
+        '403':
+          description: 'User is not authorized to delete this badge.'
         '404':
-          description: The token was not registered or the service was unknown.
-  /feeds/status:
+          description: 'The UUID does not correspond to an existing badge.'
+  '/content_categories':
```
Comment (ParryLuke) [inline api.swagger.yaml]: All endpoints for this taxonomy are missing the "content_type" property as seen in the UI. Which should be mandatory for POST requests

Comment (sam-ludlow) [reply api.swagger.yaml]: @ParryLuke I've been asking about this one for a while but got no answer. It's a bit tricky:
- It's an array so how will PUTs work for example? If you want to modify the array does the client GET the list manipulate it then PUT it back? Or would you have another endpiont?
-  content_types don't have a UUID (custom ones do, built in have a machine name) example solr :
"sm_field_content_type":["custom_60",
          "advpoll",
          "blog",
          "doc",
          "manual",
          "manual_page",
          "group_status",
          "custom_1574"],

File: api.swagger.yaml [hunk 2]
```diff
@@ -2777,1020 +6702,1666 @@ components:
...
+      allOf:
+        - type: object
+          properties:
+            uuid:
+              description: 'Invotra UUID of the policy type.'
+              type: string
+              format: uuid
+              example: '01234567-89ab-cdef-1234-56789abcdef0'
+            has_children:
+
```
Comment (ParryLuke) [inline api.swagger.yaml]: missing users array (see UI)

File: api.swagger.yaml [hunk 3]
```diff
@@ -2392,172 +2392,3580 @@ paths:
...
         '404':
-          description: Unknown service.
+          description: 'The UUID does not correspond to an existing badge.'
+        '422':
+          description: 'The badge was not updated due to a Drupal validation failure or a JSON schema validation failure.'
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/badge-schema-create'
```
[the remark this answers is not in the document]
Comment (sam-ludlow) [reply api.swagger.yaml]: have split all schemas

File: api.swagger.yaml [hunk 4]
```diff
@@ -2777,1020 +6702,1666 @@ components:
...
+          example: '01234567-89ab-cdef-1234-56789abcdef0'
           type: string
           format: uuid
-          example: 01234567-89ab-cdef-1234-56789abcdef0
-        privacy_setting:
-          description: This is the privacy setting for the group.
+        published:
+          description: 'Indicates that this content type is published.'
+          type: boolean
+          example: true
```
[the remark this answers is not in the document]
Comment (sam-ludlow) [reply api.swagger.yaml]: published, five_star_rating, and enable_comments all have default values to match UI

File: api.swagger.yaml [hunk 5]
```diff
@@ -2777,1020 +6702,1666 @@ components:
...
-          type: integer
-          example: 200
-          readOnly: true
+            - 'hidden'
+            - 'closed'
+            - 'open'
+        five_star_rating:
+          description: 'Indicates that this content type has a five star rating.'
+          type: boolean
+          example: true
```
[the remark this answers is not in the document]
Comment (sam-ludlow) [reply api.swagger.yaml]: done

File: api.swagger.yaml [hunk 6]
```diff
@@ -2777,1020 +6702,1666 @@ components:
...
+          example: 'Technical communication document'
+        external_id:
+          description: 'The content type external reference id.'
           type: string
-          example: Security
-          readOnly: true
-        group_category_uuid:
-          description: The uuid of the category
+          example: 'ABC123'
+        parent_uuid:
```
[the remark this answers is not in the document]
Comment (sam-ludlow) [reply api.swagger.yaml]: content type removed parent and has children

File: api.swagger.yaml [hunk 7]
```diff
@@ -2777,1020 +6702,1666 @@ components:
...
           type: string
           format: uuid
-          example: 01234567-89ab-cdef-1234-56789abcdef0
-        privacy_setting:
-          description: This is the privacy setting for the group.
+        published:
+          description: 'Indicates that this content type is published.'
+          type: boolean
+          example: true
+        weight:
```
[the remark this answers is not in the document]
Comment (sam-ludlow) [reply api.swagger.yaml]: content type removed weight

File: api.swagger.yaml [hunk 8]
```diff
@@ -2777,1020 +6702,1666 @@ components:
...
+          example: 'Staff Welfare'
+        description:
+          description: 'Description of the idea category.'
+          type: string
+          example: 'Ideas that may prevent loss of life or injury to staff'
+        external_id:
+          description: 'The idea category external reference id.'
+          type: string
+          example: 'ABC123'
+        parent_uuid:
```
[the remark this answers is not in the document]
Comment (sam-ludlow) [reply api.swagger.yaml]: idea category removed parent and has_children

File: api.swagger.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (sam-ludlow) [reply api.swagger.yaml]: query category removed has_children and parent_uuid

[the remark this answers is not in the document]
Comment (sam-ludlow) [reply api.swagger.yaml]: highlights removed has_children and parent_uuid

[the remark this answers is not in the document]
Comment (sam-ludlow) [reply api.swagger.yaml]: added property /search_highlights.highlight_list_uuid

Comment (sam-ludlow) [reply api.swagger.yaml]: Have added the users array

Comment (ParryLuke) [inline api.swagger.yaml]: typo

File: api.swagger.yaml (hunk 1 above)
Comment (sam-ludlow) [reply api.swagger.yaml]: added

File: api.swagger.yaml (hunk 2 above)
Comment (sam-ludlow) [reply api.swagger.yaml]: I can't spot the typo?

Comment (sam-ludlow) [reply api.swagger.yaml]: sorted

**Pull request merged**